tests/test_utils.py

```diff
@@ -108,24 +107,152 @@ def test_get_zuul_yaml_files_skip():
     assert len(zuulcilint_utils.get_zuul_yaml_files(tmp_path)["bad_yaml"]) == 1
 
 
-def test_get_jobs_from_zuul_yaml():
-    """Test that get_jobs_from_zuul_yaml() returns a list of jobs."""
+def test_get_zuul_object_from_yaml_job():
+    """Test that get_zuul_object_from_yaml() returns a list of jobs."""
     tmp_path = setup_zuul_job_yaml()
-    jobs = zuulcilint_utils.get_jobs_from_zuul_yaml(tmp_path / "job.yaml")
+    jobs = zuulcilint_utils.get_zuul_object_from_yaml(ZuulObject.JOB, tmp_path / "job.yaml")
     assert len(jobs) == 1
     assert jobs[0]["job"]["name"] == "test-job"
 
 
-def test_get_jobs_from_zuul_yaml_invalid_yaml():
+def test_get_zuul_object_from_yaml_nodeset():
+    """Test that get_zuul_object_from_yaml() returns a list of nodesets."""
+    tmp_path = setup_zuul_job_yaml()
+    with pathlib.Path.open(tmp_path / "job.yaml", "w", encoding="utf-8") as f:
```

Comment:
Why are you writing to a file named "job.yaml"? Shouldn't it be called "nodeset.yaml", for example?

Reply:
Updated setup to dynamically create files and objects